### DISCUSSION TITLE
AssertionError: `num_effective_subcarriers` must be an integer multiple of `num_tx`*`num_streams_per_tx`.

### DISCUSSION THREAD

**Discussion by marxwolf** [NVlabs/sionna#96] [Q&A] [answered]
Hi, I am testing the tutorial, https://nvlabs.github.io/sionna/examples/Realistic_Multiuser_MIMO_Simulations.html
When I set `num_ut=100` in **cell 3**, the execution of **cell 6** throws an error out. I search the `num_effective_subcarriers`, it is `self._mask.shape[3]`. But in this tutorial, we don't explicitly set the PilotPattern, or mask. 

I didn't change other parameters, What am I supposed to do to solve this problem?

**Comment by jhoydis** [comment, marked as answer]
You are implicitly setting the [KroneckerPilotPattern](https://nvlabs.github.io/sionna/api/ofdm.html#sionna.ofdm.KroneckerPilotPattern) in this example. 
```
rg = ResourceGrid(num_ofdm_symbols=14,
                  fft_size=128,
                  subcarrier_spacing=30e3,
                  num_tx=num_ut,
                  num_streams_per_tx=num_streams_per_tx,
                  cyclic_prefix_length=20,
                  pilot_pattern="kronecker",
                  pilot_ofdm_symbol_indices=[2,11])
rg.show();
```


The problem with 100 UTs is that this number is not compatible with the FFT size and this pilot pattern. You can try with any ``fft_size`` that is a multiple of 100. However, even if you do it, the performance won't be great because each UT sends a pilot on every 100th subcarrier. This is not sufficient for channel estimation in most cases.
